src/common/stream.cc

```diff
@@ -13,16 +13,28 @@
 
 FileBuffer::FileBuffer(const char* fileName)
     : fileName_(fileName)
+    , fd_(-1)
 {
     init(fileName);
 }
 
 FileBuffer::FileBuffer(const std::string& fileName)
     : fileName_(fileName)
+    , fd_(-1)
 {
     init(fileName.c_str());
 }
 
+FileBuffer::~FileBuffer()
+{
+    if( -1 != fd_ ) {
```

Comment:
I did not see when are you setting the fd in the `FileBuffer` ? I would expect this if to always fail